packages/realm/src/Realm.ts

```diff
@@ -511,16 +511,24 @@ export class Realm {
       );
       const table = binding.Helpers.getTable(this.internal, objectSchema.tableKey);
       table.removeObject(obj.key);
+    } else if (subject instanceof List) {
+      const internalList = subject.internal;
+      internalList.removeAll();
```

Comment:
Little need for a local variable and [the existing SDK called deleteAll](https://github.com/realm/realm-js/blob/master/src/js_realm.hpp#L1408):

```suggestion
      subject.internal.deleteAll();
```

Reply:
Done.  Also did the same for Results